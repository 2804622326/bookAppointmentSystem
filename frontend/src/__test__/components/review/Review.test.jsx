import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Review from '../../../components/review/Review.jsx';
import { UserType } from '../../../components/utils/utilities.js';

const review = {
  patientName: 'Jane',
  veterinarianName: 'Bob',
  patientImage: 'p.jpg',
  veterinarianImage: 'v.jpg',
  stars: 4,
  feedback: 'nice!',
};

test('displays patient review for vet', () => {
  const { asFragment } = render(<Review review={review} userType={UserType.VET} />);
  expect(asFragment()).toMatchSnapshot();
});

test('displays vet review for patient', () => {
  const { asFragment } = render(<Review review={review} userType={UserType.PATIENT} />);
  expect(asFragment()).toMatchSnapshot();
});
