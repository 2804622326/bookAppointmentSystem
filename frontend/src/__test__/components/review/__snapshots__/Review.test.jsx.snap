// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays patient review for vet 1`] = `
<DocumentFragment>
  <div
    class="mb-4"
  >
    <div
      class="d-flex align-item-center me-5"
    >
      <img
        alt="User photo"
        class="card-img user-image"
        src="data:image/png;base64,p.jpg"
      />
      <div>
        <div>
          <h5
            class="title ms-3"
          >
            <span
              class="me-2 ms-2"
            >
              <svg
                color="#ffc107"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 193, 7);"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                />
              </svg>
              <svg
                color="#ffc107"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 193, 7);"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                />
              </svg>
              <svg
                color="#ffc107"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 193, 7);"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                />
              </svg>
              <svg
                color="#ffc107"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 193, 7);"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                />
              </svg>
              <svg
                color="#ffc107"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 193, 7);"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
                />
              </svg>
            </span>
          </h5>
        </div>
        <div
          class="mt-4"
        >
          <p
            class="review-text ms-4"
          >
            nice!
          </p>
        </div>
        <div>
          <p
            class="ms-4"
          >
            Reviewed by: Jane
          </p>
        </div>
      </div>
    </div>
    <hr />
  </div>
</DocumentFragment>
`;

exports[`displays vet review for patient 1`] = `
<DocumentFragment>
  <div
    class="mb-4"
  >
    <div
      class="d-flex align-item-center me-5"
    >
      <img
        alt="User photo"
        class="card-img user-image"
        src="data:image/png;base64,v.jpg"
      />
      <div>
        <div>
          <h5
            class="title ms-3"
          >
            <span
              class="me-2 ms-2"
            >
              <svg
                color="#ffc107"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 193, 7);"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                />
              </svg>
              <svg
                color="#ffc107"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 193, 7);"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                />
              </svg>
              <svg
                color="#ffc107"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 193, 7);"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                />
              </svg>
              <svg
                color="#ffc107"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 193, 7);"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                />
              </svg>
              <svg
                color="#ffc107"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 193, 7);"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
                />
              </svg>
            </span>
          </h5>
        </div>
        <div
          class="mt-4"
        >
          <p
            class="review-text ms-4"
          >
            nice!
          </p>
        </div>
        <div>
          <p
            class="ms-4"
          >
            You rated Dr. Bob
          </p>
        </div>
      </div>
    </div>
    <hr />
  </div>
</DocumentFragment>
`;
